src/modules/job-manager/job.c

```diff
@@ -61,7 +61,7 @@ struct job *job_create (void)
     if (!(job = calloc (1, sizeof (*job))))
         return NULL;
     if (!(job->events = bitmap_alloc0 (EVENTS_BITMAP_SIZE))) {
-        free (job);
+        job_decref (job);
         return NULL;
     }
     job->refcount = 1;
```

Comment:
`free()` clobbers errno but `job_decref()` (the destructor for the `struct job` we are creating) does not.

Reply:
We do that quite frequently in constructors that allocate a structure with `calloc()` but if something goes wrong in the constructor we pass to the destructor instead of simply `free()`. (Though admittedly I didn't look that closely here)

Reply:
ohhhhh i'm sorry, i was thinking `json_decref()` .... this is `job_decref()`.   all is good, you can set MWP again :-)

Reply:
Glad you're keeping an eye on things @chu11!  Thanks guys.